### ISSUE TITLE
SciDB hangs when trying to import a 1.5 GB matrix from R

### ISSUE THREAD

**Issue by kaneplusplus** [Paradigm4/SciDBR#13]
The following causes R to hang:

``` r
require(scidb)
scidbconnect()
x <- rbind(matrix(rnorm(1e8, sd = 0.3), ncol = 5),
           matrix(rnorm(1e8, mean = 1, sd = 0.3), ncol = 5))
x <- as.scidb(x)
```

**Comment by bwlewis** [comment]
This bug is still popping up, in even worse forms by crashing. The bugs are internal to RCurl, andd occur when downloading large objects. Consider this 5GB download example (on a machine with plenty of RAM to handle it):

```
*** caught segfault ***
address 0x7f9d68f0e000, cause 'memory not mapped'

Traceback:
 1: .Call("R_curl_easy_perform", curl, .opts, isProtected, .encoding,     PACKAGE = "RCurl")
 2: curlPerform(curl = curl, .opts = opts, .encoding = .encoding)
 3: getURL(url, ..., write = getNativeSymbolInfo("R_curl_write_binary_data")$address,     file = .buf@ref, curl = curl, .opts = .opts)
 4: getBinaryURL(r, .opts = c(curlopts(), list(ssl.verifyhost = as.integer(options("scidb.verifyhost")),     ssl.verifypeer = 0, noprogress = !interrupt, progressfunction = curl_signal_trap)))
 5: doTryCatch(return(expr), name, parentenv, handler)
 6: tryCatchOne(expr, names, parentenv, handlers[[1L]])
 7: tryCatchList(expr, classes, parentenv, handlers)
 8: tryCatch({    getBinaryURL(r, .opts = c(curlopts(), list(ssl.verifyhost = as.integer(options("scidb.verifyhost")),         ssl.verifypeer = 0, noprogress = !interrupt, progressfunction = curl_signal_trap)))}, error = function(e) {    GET("/release_session", list(id = sessionid))    sigint(SIG_DFL)    stop("canceled")})
 9: scidb_unpack_to_dataframe(query, ...)
10: 
```

**Comment by bwlewis** [comment]
Fixed in https://github.com/Paradigm4/SciDBR/commit/e0241410e530be1832229f39db0bd034daa89ef2

**Comment by bwlewis** [comment]
The crashing is fixed.

But performance on the R side is awful. For example, I created a 4GB vector:

```
iquery -naq "store(build(<x:double>[i=1:500000000,1000000,0],random()), x)"
```

And then proceed to download that into R on a single computer with 16GB total RAM:
```r
t1 = proc.time()
x = scidb("x")[ ]
proc.time() - t1
```

The SciDB query + data transfer portion of this process took about 56 seconds (with `options(scidb.stream=TRUE)`).

Assembling this into a vector in R frequently dipped into swap and took over 5 minutes before I gave up.
